accio-cache/src/main/java/io/accio/cache/CacheManager.java

```diff
@@ -96,58 +101,82 @@ public CacheManager(
     {
         CatalogSchemaTableName catalogSchemaTableName = catalogSchemaTableName(mdl.getCatalog(), mdl.getSchema(), cacheInfo.getName());
         Optional<Task> taskOptional = Optional.ofNullable(tasks.get(catalogSchemaTableName));
         if (taskOptional.isPresent() && taskOptional.get().getTaskInfo().inProgress()) {
             throw new AccioException(GENERIC_USER_ERROR, format("cache is already running; catalogName: %s, schemaName: %s, tableName: %s", mdl.getCatalog(), mdl.getSchema(), cacheInfo.getName()));
         }
         removeCacheIfExist(catalogSchemaTableName);
-        return doCache(mdl, cacheInfo);
+        return doCache(mdl, cacheInfo, taskInfo);
     }
 
-    private CompletableFuture<Void> handleCache(AccioMDL mdl, CacheInfo cacheInfo)
+    private CompletableFuture<Void> handleCache(AccioMDL mdl, CacheInfo cacheInfo, TaskInfo taskInfo)
     {
-        return refreshCache(mdl, cacheInfo)
+        CatalogSchemaTableName catalogSchemaTableName = new CatalogSchemaTableName(mdl.getCatalog(), mdl.getSchema(), cacheInfo.getName());
+        String duckdbTableName = format("%s_%s", cacheInfo.getName(), randomUUID().toString().replace("-", ""));
+        long createTime = currentTimeMillis();
+        return refreshCache(mdl, cacheInfo, taskInfo)
                 .thenRun(() -> {
                     if (cacheInfo.getRefreshTime().toMillis() > 0) {
                         cacheScheduledFutures.put(
-                                new CatalogSchemaTableName(mdl.getCatalog(), mdl.getSchema(), cacheInfo.getName()),
+                                catalogSchemaTableName,
                                 refreshExecutor.scheduleWithFixedDelay(
                                         () -> createTask(mdl, cacheInfo).join(),
                                         cacheInfo.getRefreshTime().toMillis(),
                                         cacheInfo.getRefreshTime().toMillis(),
                                         MILLISECONDS));
                     }
+                })
+                .exceptionally(e -> {
+                    // If the cache fails because DuckDB doesn't have sufficient memory, we'll attempt to retry it later.
+                    if (e.getCause() instanceof AccioException && EXCEEDED_GLOBAL_MEMORY_LIMIT.toErrorCode().equals(((AccioException) e.getCause()).getErrorCode())) {
+                        retryScheduledFutures.put(
+                                catalogSchemaTableName,
+                                retryExecutor.schedule(
+                                        () -> createTask(mdl, cacheInfo).join(),
+                                        duckdbConfig.getCacheTaskRetryDelay(),
+                                        SECONDS));
+                    }
+                    duckdbClient.dropTableQuietly(duckdbTableName);
+                    String errMsg = format("Failed to do cache for cacheInfo %s; caused by %s", cacheInfo.getName(), e.getMessage());
```

Comment:
Let's add more information for it.
```suggestion
                    String errMsg = format("Failed to do cache for cacheInfo %s; caused by %s", cacheInfo.getName(), e.getMessage());
                    // If the cache fails because DuckDB doesn't have sufficient memory, we'll attempt to retry it later.
                    if (e.getCause() instanceof AccioException && EXCEEDED_GLOBAL_MEMORY_LIMIT.toErrorCode().equals(((AccioException) e.getCause()).getErrorCode())) {
                        retryScheduledFutures.put(
                                catalogSchemaTableName,
                                retryExecutor.schedule(
                                        () -> createTask(mdl, cacheInfo).join(),
                                        duckdbConfig.getCacheTaskRetryDelay(),
                                        SECONDS));
                        errMsg += "; will retry after " + duckdbConfig.getCacheTaskRetryDelay() + " seconds";
                    }
                    duckdbClient.dropTableQuietly(duckdbTableName);
```